src/dosbox.cpp

```diff
@@ -812,34 +812,92 @@ void DOSBOX_Init()
 	pstring->SetDeprecatedWithAlternateValue("486_prefetch", "486");
 	pstring->SetDeprecatedWithAlternateValue("pentium_slow", "pentium");
 
-	pmulti_remain = secprop->AddMultiValRemain("cycles",
-	                                           deprecated_but_allowed,
-	                                           " ");
+	// Legacy `cycles` setting
+	pmulti_remain = secprop->AddMultiValRemain("cycles", deprecated_but_allowed, " ");
 	pmulti_remain->Set_help(
-	        "Number of instructions DOSBox tries to emulate per millisecond\n"
-	        "('auto' by default). Setting this value too high may result in sound drop-outs\n"
-	        "and lags.\n"
-	        "  auto:            Try to guess what a game needs. It usually works, but can\n"
-	        "                   fail with certain games.\n"
-	        "  fixed <number>:  Set a fixed number of cycles. This is what you usually\n"
-	        "                   need if 'auto' fails (e.g. 'fixed 4000').\n"
-	        "  max:             Allocate as much cycles as your computer is able to handle.");
+	        "The 'cycles' setting is deprecated but still accepted; please use the\n"
+	        "'cpu_cycles', 'cpu_cycles_protected' and 'cpu_throttle' settings instead as\n"
+	        "support will be removed in the future.");
```

Comment:
This text does not appear in the config file after building this branch and deleting my old config.  Only the new cycles settings are shown.  That's probably your intent but no one is going to see this message.

Reply:
Yes, that's 100% intentional. The message will be seen in the logs if you set `cycles = <whatever>` or when you run `config -h cycles`, but it's hidden when writing the config file. I see this as the path forward for legacy settings we must still accept because of backward compatibility, but we effectively want to sweep them under the carpet and only reluctantly tell the user about their existence 😆 That's what the new `DeprecatedButAllowed` config setting type does.

<img width="1067" alt="image" src="https://github.com/dosbox-staging/dosbox-staging/assets/698770/99e80e58-d6fa-46a4-940a-927a69303b8d">